homeassistant/components/hydrawise/__init__.py

```diff
@@ -47,13 +59,18 @@
 
 def setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Hunter Hydrawise component."""
+    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in config:
+        # We were set up using config flow and not YAML.
+        return True
+
     conf = config[DOMAIN]
     access_token = conf[CONF_ACCESS_TOKEN]
     scan_interval = conf.get(CONF_SCAN_INTERVAL)
```

Comment:
YAML should be imported and deprecated (search for `deprecated_yaml` in the code)

Reply:
That makes sense, but should it wait until:

1. We have feature parity between YAML and Config Flow?
2. We have some confidence of what the correct `breaks_in_ha_version` value should be?  (Between initial review lag and my own lag in response, I'm not sure I could give a reasonable estimate for when this would actually be merged)

Reply:
Please add a comment in the code to that effect.

```python
    if DOMAIN not in config:
        # We were set up using config flow and not YAML.
        return True

    # YAML config uses the old library and is kept until feature parity is achieved
    conf = config[DOMAIN]
```

Reply:
Done.